src/backend/joanie/payment/backends/base.py

```diff
@@ -174,29 +174,27 @@ def _do_on_payment_failure(cls, order, installment_id):
         cls._send_mail_refused_debit(order, installment_id)
 
     @staticmethod
-    def _do_on_refund(amount, invoice, refund_reference):
+    def _do_on_refund(
+        amount, invoice, refund_reference, installment_id, is_transaction_canceled=False
+    ):
```

Comment:
IMO there is an issue with this method. I don't understand how you can work on refunding installment, etc while still having an order not in refund state and at the end the order is still not in `refund` state. Probably I'm missing something.